litmus-portal/graphql-server/pkg/chaos-workflow/handler/handler.go

```diff
@@ -400,78 +399,151 @@ func QueryWorkflowRuns(input model.GetWorkflowRunsInput) (*model.GetWorkflowsOut
+			matchWfNameStage := bson.D{
+				{"$match", bson.D{
+					{"workflow_name", bson.D{
+						{"$regex", workflowInput.Filter.WorkflowName},
+					}},
+				}},
 			}
-			result = append(result, &newChaosWorkflows)
+			pipeline = append(pipeline, matchWfNameStage)
+		}
+
+		// Filtering based on cluster name
+		if workflowInput.Filter.ClusterName != nil && *workflowInput.Filter.ClusterName != "All" && *workflowInput.Filter.ClusterName != "" {
+			matchClusterStage := bson.D{
+				{"$match", bson.D{
+					{"cluster_name", workflowInput.Filter.ClusterName},
+				}},
+			}
+			pipeline = append(pipeline, matchClusterStage)
 		}
 	}
 
-	return result, nil
-}
+	var sortStage bson.D
 
-// QueryListWorkflow returns all the workflows present in the given project
-func QueryListWorkflow(project_id string, workflowIds []*string) ([]*model.Workflow, error) {
-	var query bson.D
-	if len(workflowIds) != 0 {
-		query = bson.D{
-			{"project_id", project_id},
-			{"workflow_id", bson.M{"$in": workflowIds}},
+	switch {
+	case workflowInput.Sort != nil && workflowInput.Sort.Field == model.WorkflowSortingFieldName:
+		// Sorting based on WorkflowName time
+		if workflowInput.Sort.Descending != nil && *workflowInput.Sort.Descending {
+			sortStage = bson.D{
+				{"$sort", bson.D{
+					{"workflow_name", -1},
+				}},
+			}
+		} else {
+			sortStage = bson.D{
+				{"$sort", bson.D{
+					{"workflow_name", 1},
+				}},
+			}
 		}
-	} else {
-		query = bson.D{
-			{"project_id", project_id},
+	default:
+		// Default sorting: sorts it by LastUpdated time in descending order
+		sortStage = bson.D{
+			{"$sort", bson.D{
+				{"updated_at", -1},
+			}},
 		}
 	}
-	chaosWorkflows, err := dbOperationsWorkflow.GetWorkflows(query)
+
+	// Pagination
+	paginatedWorkflows := bson.A{
+		sortStage,
+	}
+
+	if workflowInput.Pagination != nil {
+		paginationSkipStage := bson.D{
+			{"$skip", workflowInput.Pagination.Page * workflowInput.Pagination.Limit},
+		}
+		paginationLimitStage := bson.D{
+			{"$limit", workflowInput.Pagination.Limit},
+		}
+
+		paginatedWorkflows = append(paginatedWorkflows, paginationSkipStage, paginationLimitStage)
+	}
+
+	// Add two stages where we first count the number of filtered workflow and then paginate the results
+	facetStage := bson.D{
+		{"$facet", bson.D{
+			{"total_filtered_workflows", bson.A{
+				bson.D{{"$count", "count"}},
+			}},
+			{"scheduled_workflows", paginatedWorkflows},
+		}},
+	}
+	pipeline = append(pipeline, facetStage)
+
+	// Call aggregation on pipeline
+	workflowsCursor, err := dbOperationsWorkflow.GetAggregateWorkflows(pipeline)
 
 	if err != nil {
 		return nil, err
 	}
+
 	var result []*model.Workflow
-	for _, workflow := range chaosWorkflows {
+
+	var workflows []dbSchemaWorkflow.AggregatedWorkflows
+
+	if err = workflowsCursor.All(context.Background(), &workflows); err != nil {
+		return &model.ListWorkflowsOutput{
+			TotalNoOfWorkflows: 0,
+			Workflows:          result,
+		}, nil
+	}
+
```

Comment:
Please add a check condition for `workflows` here to catch out of bound issue

Reply:
Done